File: Interpreter/src/model/VarExpresion.java
package model;

import DataStructure.IntHeap;
import DataStructure.MyIntDict;
import ExceptionHandling.MyException;

public class VarExpresion extends Expresion {
    String id;

    public VarExpresion(String _id){
        id = _id;
    }

    @Override
    int eval(MyIntDict<String, Integer> tab, IntHeap heap) throws MyException {
        try {
            return tab.get(id);
        }
        catch (MyException ex){
            throw ex;
        }
    }
    @Override
    public String toString() {
        return id;
    }
}

File: map/Interpreter2.0/model/VarExpresion.cs
using System;
using DataStructure;
namespace models
{
    public class VarExpresion : Expression
    {
        String id;

        public VarExpresion(String _id){
            id = _id;
        }

        public override int eval(MyIntDict<String, int> tab) {
            try {
                return tab.get(id);
            }
            catch (MyException.MyException ex){
                throw ex;
            }
        }
 
        override public String ToString() {
            return id;
        } 
    }
}